src/sagemaker/local/entities.py

```diff
@@ -38,6 +38,136 @@
 HEALTH_CHECK_TIMEOUT_LIMIT = 120
 
 
+class _LocalProcessingJob(object):
+    """Defines and starts a local processing job."""
+
+    _STARTING = "Starting"
+    _PROCESSING = "Processing"
+    _COMPLETED = "Completed"
+    _states = ["Starting", "Processing", "Completed"]
+
+    def __init__(self, container):
+        """
+        Args:
+            container: the local container object.
+        """
+        self.container = container
+        self.state = "Created"
+        self.start_time = None
+        self.end_time = None
+        self.processing_job_name = ""
+        self.processing_inputs = None
+        self.processing_output_config = None
+        self.environment = None
+
+    def start(self, processing_inputs, processing_output_config, environment, processing_job_name):
+        """
+        Args:
+            processing_inputs: The processing input configuration.
+            processing_output_config: The processing input configuration.
+            environment: The collection of environment variables passed to the job.
+            processing_job_name: The processing job name.
+        """
+
+        for item in processing_inputs:
+            if item["S3Input"]:
+                data_uri = item["S3Input"]["S3Uri"]
+            else:
+                raise ValueError("Processing input must have a valid ['S3Input']")
```

Comment:
with dataset definitions as valid input, may need to consider handling that.

https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/processing.py#L1026-L1039

Reply:
Done.